scripts/health_check_links.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+
+# This script verifies that no provided links in the repo are broken. It does
+# so by making a HTTP request to each website and looking at the status code of
+# the response.
+#
+# If the request responds with 5xx the script terminates with a status code of
+# 1, meaning a link is broken. 3xx and 4xx responses are treated as warnings
+# and are simply logged, because they do not guarantee that there is something
+# wrong with the requested website. The status code 000 is also treated as a
+# warning because the status code alone does not specify where the problem
+# lies, only that there is a problem, read more here: https://tinyurl.com/superuser-status-code-000
+#
+### Dependencies
+# - curl
+# - GNU grep
+# - GNU parallel
+#
+### Usage
+#
+#  /bin/bash ./health_check_links.sh
+#
+### Improvements to be made
+# - Be able to Blacklist files to be ignored
+# - Output the actual problem if the response status code was 000 (see link
+#   above for more info)
+# - Do not use GNU grep
+#
+# Author: http://github.com/simeg
+# License: MIT
+#
+
+readonly PARALLEL_JOBS_COUNT=200
+
+echo "🔎 Finding markdown files.."
+readonly MARKDOWN_FILES_STR=$(find "$(PWD)"/.. -type f -name "*.md")
+readonly MARKDOWN_FILES_ARR=(${MARKDOWN_FILES_STR// / })
+echo "Found [" ${#MARKDOWN_FILES_ARR[@]} "] files"
+
+echo "🔬 Parsing URLs.."
+URL_ARR=()
+for FILE in "${MARKDOWN_FILES_ARR[@]}"; do
+  # Parse URL
+  URLS_STR=$(cat $FILE | ggrep -o -E 'https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)')
```

Comment:
http instead of https as there might be some of those?

Reply:
The `s?` in `https?` means zero or one s. So it finds http or https.

Reply:
Ah yes, should have read it more carefully, thanks.